addOns/network/src/main/java/org/zaproxy/addon/network/ExtensionNetwork.java

```diff
@@ -122,15 +138,25 @@
 import org.zaproxy.zap.extension.api.ApiImplementor;
 import org.zaproxy.zap.extension.brk.ExtensionBreak;
 import org.zaproxy.zap.network.HttpSenderImpl;
+import org.zaproxy.zap.utils.DisplayUtils;
 import org.zaproxy.zap.utils.ZapPortNumberSpinner;
+import org.zaproxy.zap.view.ZapTable;
+import org.zaproxy.zap.view.popup.PopupMenuHttpMessageContainer;
+import org.zaproxy.zap.view.renderer.DateFormatStringValue;
 
 public class ExtensionNetwork extends ExtensionAdaptor implements CommandLineListener {
 
     private static final Logger LOGGER = LogManager.getLogger(ExtensionNetwork.class);
+    private static final ImageIcon RATE_LIMIT_ICON;
```

Comment:
Probably better to move all the rate limiter code to its own class to keep the extension tidier.

Reply:
Then get tabs/icons/whatever from there statically?

Reply:
No statically, it would call the required methods and let the other class add/removes the things as needed.